sources/web/datalab/polymer/test/table-details-test.ts

```diff
@@ -0,0 +1,132 @@
+      projectId: 'pid',
+      tableId: 'tid',
+    },
+    type: 'table',
+  };
+
+  /**
+   * Rows must be recreated on each test with the fixture, to avoid state leakage.
+   */
+  beforeEach(() => {
+    GapiManager.getBigqueryTableDetails = (pid, did, tid) => {
+      assert(!!pid && !!did && !!tid,
+          'getTableDetails should be called with project, dataset, and table');
+
+      const request = {
+        result: mockTable,
+      };
+      return Promise.resolve(request) as any;
+    };
+
+    testFixture = fixture('table-details-fixture');
+    testFixture.tableId = '';
+    Polymer.dom.flush();
+  });
+
+  it('displays an empty element if no table is provided', () => {
+    assert(!testFixture.$.placeholder.hidden, 'placeholder should be shown');
+    assert(testFixture.$.container.hidden, 'container should be hidden');
+    assert(testFixture.$.placeholder.querySelector('paper-spinner'),
+        'spinner should be hidden if no table is loading');
+  });
+
+  it('loads the table given its id and gets its details', (done: () => null) => {
+    testFixture.addEventListener('_table-changed', () => {
+      Polymer.dom.flush();
+
+      assert(JSON.stringify((testFixture as any)._table) === JSON.stringify(mockTable),
+          'element should have fetched table info');
+
+      assert(testFixture.$.tableId.innerText === mockTable.tableReference.tableId,
+            'table id should be parsed');
+      assert(testFixture.$.projectId.innerText === mockTable.tableReference.projectId,
+            'project id should be parsed');
+      assert(testFixture.$.datasetId.innerText === mockTable.tableReference.datasetId,
+            'dataset id should be parsed');
+      assert(testFixture.$.tableSize.innerText === '1023.00 KB', 'should see readable table size');
+      assert(testFixture.$.longTermTableSize.innerText === '50.00 MB',
+          'should see readable long term table size');
+      assert(testFixture.$.numRows.innerText === '1,234,567,890',
+          'should see comma-separated number of rows');
+      assert(testFixture.$.creationTime.innerText === '7/31/2017, 3:47:51 PM',
+          'should parse timestamp into readable text');
+      assert(testFixture.$.lastModifiedTime.innerText === '7/31/2017, 3:47:51 PM',
+          'should parse timestamp into readable text');
+      assert(testFixture.$.location.innerText === mockTable.location, 'should see location');
+
+      assert(testFixture.$.schema.rows.length === mockTable.schema.fields.length,
+          'unexpected number of schema rows');
+
+      const rows = testFixture.$.schema.rows;
+      mockTable.schema.fields.forEach((field, i) => {
+        assert(rows[i].children[0].innerText === field.name, 'field name not matching');
+        assert(rows[i].children[1].innerText === field.type, 'field type not matching');
+        assert(rows[i].children[2].innerText === (field.mode || 'NULLABLE'),
+            'field mode not matching');
+      });
+
+      done();
```

Comment:
Right. I was noticing that all of the asserts are within the callback block, so if for some reason the event never fired, and the callback never got called, no assertions would fail. Thus I just wanted to be sure that the test fails if done() is not called. Please tell me the test fails if you comment out the call to done().